### PR TITLE
Issue/17 log security

### LINKED ISSUE #17: Removal of logs
Let's also add functionality where we add to the php code for `/phpbb/log/log.php`'s `delete()` method to prevent it from deleting the log entries (silently, don't error) and email the log entries attempted to be deleted to the security email. We'll need a core event in this method to be able to do this though.

### PR DESCRIPTION
**Pull request by VSEphpbb**
Closes #17

### CODE REVIEW

**Comment by VSEphpbb** [comment]
An option/config for this would probably be good, no?

**Comment by michaelcullum** [comment]
No, as then they could just flick the switch and disable logs. This requires them to have file access (to disable the extension).

File: event/listener.php [hunk 1]
```diff
@@ -116,6 +117,45 @@ public function set_team_password_configs($event)
 	}
 
 	/**
+	 * Prevent deletion of Admin logs and notify board security contact
```
Comment (michaelcullum) [inline event/listener.php]: capital A

File: event/listener.php [hunk 2]
```diff
@@ -116,6 +117,45 @@ public function set_team_password_configs($event)
...
+	 *
+	 * @param object $event The event object
+	 * @return null
+	 * @access public
+	 */
+	public function admin_logs_security($event)
+	{
+		if ($event['mode'] == 'admin')
```
Comment (michaelcullum) [inline event/listener.php]: We want it for all logs, not just admin logs.

File: language/en/email/acp_logs.txt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Subject: {SITENAME} Admin Log Deletion Attempted
+
+The user "{USERNAME}" recently attempted to delete the following Admin logs at {TIME} from the IP address {IP_ADDRESS}:
```
Comment (michaelcullum) [inline language/en/email/acp_logs.txt]: capital A

**Comment by VSEphpbb** [comment]
But only founders can access the settings. No option makes deleting logs a real PITA.

File: event/listener.php (hunk 2 above)
Comment (VSEphpbb) [inline event/listener.php]: All, really?  Or just admin and users

**Comment by VSEphpbb** [comment]
Updated

File: language/en/email/acp_logs.txt [hunk 4]
```diff
@@ -0,0 +1,7 @@
+Subject: {SITENAME} {LOG_MODE} log deletion attempted
+
+The user "{USERNAME}" recently attempted to delete the following Admin logs at {TIME} from the IP address {IP_ADDRESS}:
```
Comment (nickvergessen) [inline language/en/email/acp_logs.txt]: Its not admin logs only

Comment (VSEphpbb) [inline language/en/email/acp_logs.txt]: oops thanks

**Pull request merged**